integration/common/openlineage/common/provider/dbt.py

```diff
@@ -0,0 +1,202 @@
+import json
+import yaml
+import os
+import uuid
+from typing import List, Tuple, Dict, Optional
+
+import attr
+
+from openlineage.client.facet import DataSourceDatasetFacet, SchemaDatasetFacet, SchemaField, \
+    SqlJobFacet
+from openlineage.common import __version__
+from openlineage.client.run import RunEvent, RunState, Run, Job, Dataset
+from openlineage.common.utils import get_from_nullable_chain
+
+PRODUCER = f"openlineage-dbt/{__version__}"
+
+
+@attr.s
+class DbtRun:
+    started_at: str = attr.ib()
+    completed_at: str = attr.ib()
+    status: str = attr.ib()
+    inputs: List[Dict] = attr.ib()
+    output: Dict = attr.ib()
+    job_name: str = attr.ib()
+    namespace: str = attr.ib()
+    run_id: str = attr.ib(factory=lambda: str(uuid.uuid4()))
+
+
+@attr.s
+class DbtEvents:
+    starts: List[RunEvent] = attr.ib()
+    completes: List[RunEvent] = attr.ib()
+
+
+class DbtArtifactProcessor:
+    def __init__(self, project: str = 'dbt_project.yaml'):
+        self.dir = os.path.abspath(os.path.dirname(project))
+        self.project = self.load_yaml(project)
+        self.namespace = ""
```

Comment:
Is an empty string a reasonable namespace default?

Reply:
The namespace is set later in `extract_namespace` method. We need dbt profile to properly get the namespace, so the value is just a placeholder.